src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -145,11 +163,20 @@ public FileSystemSecurityRealm(final Path root, final int levels) {
      * @param root the root path of the identity store
      */
     public FileSystemSecurityRealm(final Path root) {
-        this(root, NameRewriter.IDENTITY_REWRITER, 2);
+        this(root, NameRewriter.IDENTITY_REWRITER, 2, true);
     }
 
-    private Path pathFor(final String name) {
+    private Path pathFor(String name) {
         assert name.codePointCount(0, name.length()) > 0;
+        if (encoded) {
+            String normalized = Normalizer.normalize(name.toLowerCase(Locale.ROOT), Normalizer.Form.NFKC)
```

Comment:
You might want to case-fold after normalization as it's possible that some un-normalized text might be weird if case-folded.